packages/core-utils/src/provider.ts

```diff
@@ -38,3 +57,200 @@ export const FallbackProvider = (config: string | FallbackProviderConfig[]) => {
   }
   return new ethers.providers.FallbackProvider(config)
 }
+
+export class GenesisJsonProvider implements AbstractProvider {
+  genesis: Genesis
+
+  constructor(genesis: string | Genesis) {
+    if (typeof genesis === 'string') {
+      this.genesis = require(genesis)
+    } else if (typeof genesis === 'object') {
+      this.genesis = genesis
+    }
+
+    if (this.genesis === null) {
+      throw new Error('Must initialize with genesis object')
+    }
+  }
+
+  async getBalance(
+    addressOrName: string,
+    blockTag?: BlockTag
+  ): Promise<BigNumber> {
+    const address = remove0x(addressOrName)
+    const account = this.genesis.alloc[address]
```

Comment:
we're using the `alloc` field here, so this seems like the genesis file from the contract deployment - i thought we wanted to test the post-surgery state dump?

Reply:
The post surgery state dump file is a genesis file, the input and the output are the same